### PR TITLE
add a periodic cluster-api-vsphere job

### PR DESCRIPTION
**Pull request by figo**
notes: not sure whether i can share a preset with cluster-api-provider-vsphere-presubmits.yaml

cc @BenTheElder  @krzyzacy @akutz @frapposelli

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cluster-api-provider-vsphere/cluster-api-provider-vsphere-ci.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+periodics:
+  #kubernetes-sigs/cluster-api-provider-vsphere:
```
Comment (figo) [inline config/jobs/kubernetes-sigs/cluster-api-provider-vsphere/cluster-api-provider-vsphere-ci.yaml]: @BenTheElder periodic job does not support repo name,  so i commented it out, but the job still need to checkout the repo, how can i do it?

Comment (BenTheElder) [reply config/jobs/kubernetes-sigs/cluster-api-provider-vsphere/cluster-api-provider-vsphere-ci.yaml]: :man_facepalming:
`extra_refs:` -- we can fix this in a follow-up

Comment (BenTheElder) [reply config/jobs/kubernetes-sigs/cluster-api-provider-vsphere/cluster-api-provider-vsphere-ci.yaml]: cc me for review and I will re-trigger this with the updated config

Comment (figo) [reply config/jobs/kubernetes-sigs/cluster-api-provider-vsphere/cluster-api-provider-vsphere-ci.yaml]: @BenTheElder  thanks a lot, cced you here: https://github.com/kubernetes/test-infra/pull/11321

**Pull request merged**